src/GraphQL/Resolvers/SourceStreamMethodResolver.cs

```diff
@@ -86,16 +109,48 @@ public SourceStreamMethodResolver(MethodInfo methodInfo, LambdaExpression instan
                         bodyExpression);
                 }
             }
+            // IAsyncEnumerable<T>
+            else if (bodyExpression.Type.IsGenericType && bodyExpression.Type.GetGenericTypeDefinition() == typeof(IAsyncEnumerable<>))
+            {
+                var innerType = bodyExpression.Type.GetGenericArguments()[0];
+                var method = _convertFromAsyncEnumerableMethodInfo.MakeGenericMethod(innerType);
+                var func = method.CreateDelegate<Func<Expression, ParameterExpression, Func<IResolveFieldContext, ValueTask<IObservable<object?>>>>>();
+                return func(bodyExpression, resolveFieldContextParameter);
+            }
 
             if (taskBodyExpression == null)
             {
-                throw new InvalidOperationException("Method must return a IObservable<T> or Task<IObservable<T>> where T is a reference type.");
+                throw new InvalidOperationException("Method must return a IObservable<T> or Task<IObservable<T>> where T is a reference type, or a IAsyncEnumerable<T> or Task<IAsyncEnumerable<T>>.");
```

Comment:
\+ `ValueTask<IObservable<T>>` + `ValueTask<IAsyncEnumerable<T>>`

Reply:
I intentionally simplified the error message... do you think it should cover that also?

Reply:
Note that generally we encourage the use of synchronous or Task-based asynchronous calls throughout the GraphQL.NET API.  `ValueTask` support was added under the hood for performance since we are often wrapping synchronous calls with `ValueTask`, but we do not have builder methods and similar with `ValueTask`-based delegates.

So when I updated the auto-registering classes, I added support for `ValueTask` if someone were to use it, but I didn't mention `ValueTask` in the exception messages.  If you think I should do so, I can.

Reply:
Better to mention ValueTask option in error messages IMO.

Reply:
Where reference constraint come from?

Reply:
> + `ValueTask<IObservable<T>>` + `ValueTask<IAsyncEnumerable<T>>`

I just did the most simple; add the two additional types to the error message.

Reply:
OK. IMO in the new PR message(s) will be changed.